### PR TITLE
Self update

### PR DESCRIPTION
**Pull request by DBX12**
Adds a selfupdate script to bitbutler as documented in #7 
Validation via SHA256 and GPG key are missing (due to missing release and missing files)

### CODE REVIEW

File: src/bitbutler.sh [hunk 1]
```diff
@@ -126,6 +126,9 @@ ${b}Commands$w
         ${y}delete$w    Delete a default reviewer
         ${y}list$w      List default reviewers
 
+    ${b}selfupdate$w
```
Comment (particleflux) [inline src/bitbutler.sh:129]: This should be in the man page as well

File: src/bitbutler.sh [hunk 2]
```diff
@@ -698,6 +701,69 @@ JSON
   esac
 }
 
+function selfupdate() {
+  REPOSITORY_SLUG="yiisoft/yii2"
```
Comment (particleflux) [inline src/bitbutler.sh:705]: Forgot some testing stuff, hu?

File: src/bitbutler.sh [hunk 3]
```diff
@@ -698,6 +701,69 @@ JSON
...
+    l "You already have the lasted version."
+    exit 0
+  fi
+
+  confirm "Do you want to update now?"
+
+  tarball_url=$(jq --raw-output '.tarball_url' <<<"${latest_tag_json}")
+  tarball_path="${SELFUPDATE_WORKING_DIR}/tarball.tar.gz"
+
+  rm -r "${SELFUPDATE_WORKING_DIR}"
```
Comment (particleflux) [inline src/bitbutler.sh:735]: I think this `rm` should have the `-f` flag as well. It avoids potential errors or inteactivity

File: Makefile [hunk 4]
```diff
@@ -8,13 +8,20 @@ INSTALL ?= install
...
 
-install: doc
+install: doc persist
 	$(INSTALL) -D -t "${DESTDIR}${DATADIR}/${PROGRAM_NAME}" src/*.sh
 	$(INSTALL) -D man/bitbutler.1 $(DESTDIR)$(MANDIR)/man1/${PROGRAM_NAME}.1
 	mkdir -p "${DESTDIR}${BINDIR}"
 	echo -e '#!/usr/bin/env bash\nBB_VENDOR_PATH="${DESTDIR}${DATADIR}/${PROGRAM_NAME}" exec ${DESTDIR}${DATADIR}/${PROGRAM_NAME}/bitbutler.sh "$$@"' > "${DESTDIR}${BINDIR}/${PROGRAM_NAME}"
 	chmod u+rwx,g+x,o+x "${DESTDIR}${BINDIR}/${PROGRAM_NAME}"
 
+persist:
```
Comment (particleflux) [inline Makefile:18]: This *should* probably include the `DESTDIR` as well, as otherwise it might lead to a broken state

**Review by particleflux — CHANGES_REQUESTED** [review]

**Comment by particleflux** [comment]
I'd probably hold off on merging this until there are real releases here. Then it can be tested against the actual releases.

Speaking of tests, it might be a good idea to add a test for this as well, though I'm not sure of that yet.

File: src/bitbutler.sh (hunk 2 above)
Comment (DBX12) [reply src/bitbutler.sh:705]: oh noes! Must have slipped in with one of the last fixes.

File: src/bitbutler.sh (hunk 1 above)
Comment (DBX12) [reply src/bitbutler.sh:129]: I've added it to the man pages

**Review by particleflux — APPROVED** [review]

File: src/bitbutler.sh [hunk 5]
```diff
@@ -698,6 +701,69 @@ JSON
...
+  fi
+
+  latest_tag_name=$(jq --raw-output '.tag_name' <<<"$latest_tag_json")
+
+  v "Current script version: ${SCRIPT_VERSION}"
+  v "Latest version is: $latest_tag_name"
+  if [[ "$latest_tag_name" != "${SCRIPT_VERSION}" ]]; then
+    l "Your version is outdated."
+  else
+    l "You already have the lasted version."
```
Comment (particleflux) [inline src/bitbutler.sh:726]: Typo: `lasted` vs `latest`

**Review by particleflux — CHANGES_REQUESTED** [review]

File: src/bitbutler.sh (hunk 5 above)
Comment (DBX12) [reply src/bitbutler.sh]: The latest version will last a long time. :D
No idea what kind of brain-autocorrect that was^^

**Review by particleflux — APPROVED** [review]

**Comment by DBX12** [comment]
I wonder if this is counted for hacktoberfest 2021 too :joy:

**Pull request merged**